src/NiftyUploader.ts

```diff
@@ -10,16 +10,36 @@ export class NiftyUploader {
     public files: Array<NiftyFile> = new Array<NiftyFile>();
     // initilize options with default options
     public options: NiftyOptions = new NiftyDefaultOptions();
+    // whether the browser support html5 file system api.
+    public isSupport: boolean = false;
 
     //Events
     public chunkSucsessEvent: NiftyEvent<{ chunk: NiftyChunk }> = new NiftyEvent();
     public chunkFailEvent: NiftyEvent<{ chunk: NiftyChunk }> = new NiftyEvent();
     public fileQueuedEvent: NiftyEvent<{ file: NiftyFile }> = new NiftyEvent();
 
-    constructor(options: NiftyOptionsParameter) {
+    // I think because of default options, options is not required for user.
+    constructor(options?: NiftyOptionsParameter) {
```

Comment:
The option parameter has tot be non optional, because for some options there is no default value. (For example the endpoint)

Reply:
I think default endpoint could be '/' . How do you feel?

Reply:
A default value for every option is a good idea. Then I can clean up all the different Options Interfaces and Classes.